lib/broccoli/ember-app.js

```diff
@@ -805,38 +864,72 @@ EmberApp.prototype._processedAppTree = function() {
   });
 };
 
+
 /**
   @private
-  @method _processedTemplatesTree
+  @method _processedFastbootAppTree
   @return
 */
-EmberApp.prototype._processedTemplatesTree = function() {
-  var addonTrees = this.addonTreesFor('templates');
+EmberApp.prototype._processedFastbootAppTree = function() {
+  var addonTrees = this.addonTreesFor('fastboot-app');
+  var mergedApp  = mergeTrees(addonTrees.concat(this._filterFastbootAppTree()).filter(Boolean), {
+    overwrite: true,
+    annotation: 'TreeMerger (fastboot app)'
+  });
+
+  return new Funnel(mergedApp, {
+    srcDir: '/',
+    destDir: this.name,
+    annotation: '_processedFastbootAppTree'
+  });
+};
+
+/**
+  Processes the templates for 'app' or any other trees (in future)
+
+  @private
+  @method _processedAnyTemplatesTree
+ */
+EmberApp.prototype._processedAnyTemplatesTree = function(destDirPath, addonType, processorType, appTemplates) {
+  var addonTrees = this.addonTreesFor(addonType);
```

Comment:
I'm concerned that this and the `destDirPath` stuff will not play nicely with ember-engines. (Haven't done the work to confirm.)